packages/office-ui-fabric-react/src/components/Button/_base/Button.base.tsx

```diff
@@ -0,0 +1,265 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  anchorProperties,
+  autobind,
+  buttonProperties,
+  getId,
+  getNativeProps,
+  customizable,
+  nullRender
+} from '../../../Utilities';
+import { hiddenContentStyle, mergeStyles } from '../../../Styling';
+import { Icon } from '../../../Icon';
+import { IButtonBaseProps, IButtonBase, IButtonBaseStyleProps, IButtonBaseStyles } from './Button.base.types';
+import {
+  classNamesFunction
+} from '../../../Utilities';
+
+const getClassNames = classNamesFunction<IButtonBaseStyleProps, IButtonBaseStyles>();
+
+@customizable('ButtonBase', ['theme'])
+export class ButtonBase extends BaseComponent<IButtonBaseProps, {}> implements IButtonBase {
+
+  public static defaultProps = {
+  };
+
+  private _classNames: {[key in keyof IButtonBaseStyles]: string };
+  private _buttonElement: HTMLElement;
+  private _labelId: string;
+  private _descriptionId: string;
+  private _ariaDescriptionId: string;
+
+  constructor(props: IButtonBaseProps, rootClassName: string) {
+    super(props);
+
+    this._labelId = getId();
+    this._descriptionId = getId();
+    this._ariaDescriptionId = getId();
+  }
+
+  public render(): JSX.Element {
+    const {
+      theme,
+      className,
+      disabled,
+      href,
+      checked,
+      primary,
+      getStyles,
+      expanded,
+      onRenderIcon = this._onRenderIcon,
+      onRenderAriaDescription = this._onRenderAriaDescription,
+      onRenderChildren = this._onRenderChildren,
+      onRenderMenuIcon = this._onRenderMenuIcon,
+      onRenderPrefix,
+      onRenderSuffix
+    } = this.props;
+
+    // Anchor tag cannot be disabled hence in disabled state rendering
+    // anchor button as normal button
+    const renderAsAnchor: boolean = !disabled && !!href;
+
+    const nativeProps = this._getNativeProps(this.props, renderAsAnchor);
+
+    const {
+      ariaDescribedBy,
+      ariaLabelledBy,
+      ariaLabel
+    } = this._getAriaProps(
+        this.props,
+        getNativeProps(this.props, [renderAsAnchor ? 'a' : 'button']
+        ));
+
+    this._classNames = getClassNames(getStyles!, { theme: theme!, className, disabled, checked, expanded, primary });
+
+    let RootType = renderAsAnchor ? 'a' : 'button';
+    return (
+      <div className={ this._classNames.root }>
```

Comment:
Do we always want this div here even when there is no prefix or suffix?

Reply:
Yeah, I struggled with this for a bit, but I think for consistency, the root should render regardless of the content of the button. This way root styles are always honored and you don't need a different set of styles for base buttons vs menu/split buttons.